app/src/main/java/com/internship/move/feature/onboarding/OnboardingFragment.kt

```diff
@@ -15,12 +15,72 @@ class OnboardingFragment : Fragment(R.layout.fragment_onboarding) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        initListeners()
+        initViewPager()
+        initNextButton()
     }
 
-    private fun initListeners() {
-        binding.onboardingBtn.setOnClickListener {
-            findNavController().navigate(OnboardingFragmentDirections.actionGlobalRegisterFragment())
+    private fun initViewPager() {
+        binding.onboardingVP2.isUserInputEnabled = false
+        binding.onboardingVP2.adapter = OnboardingItemAdapter(this, initList())
+        binding.dotsIndicator.attachTo(binding.onboardingVP2)
+    }
+
+    private fun initNextButton() {
+        binding.nextItemBtn.setOnClickListener {
+            if (binding.onboardingVP2.currentItem < binding.onboardingVP2.adapter?.itemCount?.minus(1)!!) {
+                println("Item no" + binding.onboardingVP2.currentItem)
+                if (binding.onboardingVP2.currentItem == binding.onboardingVP2.adapter?.itemCount?.minus(2)!!)
+                    binding.nextItemBtn.text = "Get started"
+                binding.onboardingVP2.currentItem = binding.onboardingVP2.currentItem + 1
+            } else {
+                findNavController().navigate(OnboardingFragmentDirections.actionGlobalRegisterFragment())
+            }
         }
     }
+
+    private fun initList(): List<OnboardingItem> {
```

Comment:
You can simplify this method by adding your items in the mutableListOf call directly:
```
private fun initList(): List<OnboardingItem> = mutableListOf(
    OnboardingItem(..),
    OnboardingItem(..),
    // etc
)
```